### ISSUE TITLE
USWDS-Site - Feature: Add a "month" text input variation option

### ISSUE THREAD

**Issue by amycole501** [uswds/uswds-site#2394]
### Is your feature request related to a problem? Please describe.

Some disabled UX research study participants have encountered issues when interacting with the dropdown/select month element inside the memorable date. Many users have found it confusing to switch from a dropdown/select for the month then pivot into a text input for the day and year. People using assistive tech and cognitively challenged people could benefit from a variation of the memorable date where all three fields are text inputs.

### Describe the solution you'd like

Create a variation of the memorable date where all three fields are text inputs. This might mean going back to the previous version of the memorable date from before the November 2022 release.

NOTE that on the [memorable date page](https://designsystem.digital.gov/components/memorable-date/) there is a link to the research conducted leading to the change in the component resulting in the month dropdown.

However, that link takes you to research about the language selector. I'm unclear if that's an issue with the link or if the narrative around that link at the bottom needs to clarify that the language selector was also impacted.

![image](https://github.com/uswds/uswds-site/assets/117778073/d1a184dd-3268-4e08-afc6-53b9ce5b9855)
 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds-site/issues) for duplicate feature requests.

**Comment by jeana-adhoc** [comment]
Hello USWDS friends. On a VA project we recently encountered issues with memorable date in a usability study we were conducting. Below is a write up about what we encountered.

## VA’s findings
During [Form Header testing](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/CAIA/accessibility/Deliverables/Veteran%20Facing%20Forms/2023-Oct-Veteran-Facing-Forms-Form-Header-accessibility-findings.md) with assistive technology users, the VA’s accessibility specialists found that the memorable date field posed particular challenges for several users due to the inconsistency between the types of fields within the fieldset (dropdown + text inputs). One user suggested having a single text input; another suggested all dropdowns.

## Quotes
Responses from our usability study that talked about the memorable date:

* “I hate these things where you have to pick the month, but have to type the day and year…I expect the same thing (in each field)...it’s be much easier if you could just type in ‘05’ (for month), it’s simpler that way. I’m old and get confused easy…In MyHealtheVet, its numerical dates. It’d be much easier if it was the same all the way through.” - P12
* “I don’t hear a number in my day. I hear month, and then day, but no numbers are going in. In the day I am trying to get a number in there.” - P13
* P23 mentioned that having to manually type in birthday was difficult and expected a drop down like how the month is.
* P24: “Day” field was not announced after selecting a month. After typing in Day it automatically places focus to year input field.

## Impact
Nearly every form on [VA.gov](http://va.gov/) gathers date information, and should be using the memorable date component. Date of birth, date of death, and other dates of significance are a key piece of information on most benefit and compensation forms.

## Our hypothesis
We posit that having multiple inputs of different types within the same fieldset was difficult for users, and believe that replacing the month dropdown with a text input will lead to a better experience for those users.

**Comment by brunerae** [comment]
@brunerae come back and separate this Issue into two: part of the descriptions talks about a Site issue

**Comment by amycole501** [comment]
I put in a separate ticket to fix the confusing link (if a fix is indeed warranted) https://github.com/uswds/uswds-site/issues/2404
